xarray/core/rolling.py

```diff
@@ -151,34 +151,22 @@ def __init__(self, obj, min_periods=None, center=False, **windows):
         """
         super(DataArrayRolling, self).__init__(obj, min_periods=min_periods,
                                                center=center, **windows)
-        self.window_indices = None
-        self.window_labels = None
 
-        self._setup_windows()
+        self.window_labels = self.obj[self.dim]
+        self._stops = np.arange(1, len(self.window_labels) + 1)
```

Comment:
can we do this only when iteration is requested? that would be a bit more efficient for the common case where iteration is not done.

My guess is that iteration is slow enough that the overhead of recreating these objects will not be noticeable.

Reply:
Done.